superset/databases/commands/update.py

```diff
@@ -75,12 +75,21 @@ def validate(self) -> None:
         if not self._model:
             raise DatabaseNotFoundError()
         database_name: Optional[str] = self._properties.get("database_name")
+        configuration_method: Optional[str] = self._properties.get(
+            "configuration_method"
+        )
         if database_name:
             # Check database_name uniqueness
             if not DatabaseDAO.validate_update_uniqueness(
                 self._model_id, database_name
             ):
                 exceptions.append(DatabaseExistsValidationError())
+        if configuration_method:
+            if ConfigurationMethod(configuration_method) not in {
```

Comment:
Have you tested this? I believe if the value is not in the enum you're going to get an error, and you'll have to catch the error instead.

Reply:
```
def test_update_database_with_invalid_configuration_method(self):
        """
        Database API: Test update
        """
        example_db = get_example_database()
        test_database = self.insert_database(
            "test-database", example_db.sqlalchemy_uri_decrypted
        )
        self.login(username="admin")
        database_data = {
            "database_name": "test-database-updated",
            "configuration_method": "BAD_FORM",
        }
        uri = f"api/v1/database/{test_database.id}"
        rv = self.client.put(uri, json=database_data)
        self.assertEqual(rv.status_code, 400)
```

THis is the test I wrote for it.

Reply:
cool. can you maybe assert the error message, too, to validate that you're getting the right error?